pkg/resource/embed.go

```diff
@@ -0,0 +1,30 @@
+package resource
+
+import (
+	"embed"
+	"io/fs"
+)
+
+type embedResources struct {
+	fs           embed.FS
+	subDirectory string
+}
+
+func (r *embedResources) Read(file string) ([]byte, error) {
+	content, err := r.fs.ReadFile(r.subDirectory + "/" + file)
+	if err == fs.ErrNotExist {
+		return nil, ErrNotExist
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	return content, nil
+}
```

Comment:
Why not closing the file ? With a defer, like that it would handle all use cases with 1 line

Reply:
Cool!